packages/use-intl/src/core/createFormatter.tsx

```diff
@@ -158,6 +158,40 @@ export default function createFormatter({
     );
   }
 
+  function dateTimeRange(
+    /** If a number is supplied, this is interpreted as a UTC timestamp. */
+    start: Date | number,
+    /** If a number is supplied, this is interpreted as a UTC timestamp. */
+    end: Date | number,
+    /** If a time zone is supplied, the `value` is converted to that time zone.
+     * Otherwise the user time zone will be used. */
+    formatOrOptions?: string | DateTimeFormatOptions
+  ) {
+    return getFormattedValue(
+      [start, end],
```

Comment:
This argument is used by `getFormattedValue` to produce a fallback value in case formatting fails (e.g. due to an unknown format that is supplied).

Maybe we can add a test here to verify the output:

```tsx
  it('returns a reasonable fallback if an invalid format is provided', () => {
    const formatter = createFormatter({
      locale: 'en',
      timeZone: 'Europe/Berlin'
    });
    expect(
      formatter.dateTimeRange(
        new Date(2007, 0, 10, 10, 0, 0),
        new Date(2008, 0, 10, 11, 0, 0),
        'unknown'
      )
    ).toBe('TODO');
  });
```

Currently, on my machine it outputs:

```
Wed Jan 10 2007 10:00:00 GMT+0100 (Mitteleuropäische Normalzeit),Thu Jan 10 2008 11:00:00 GMT+0100 (Mitteleuropäische Normalzeit)
```

It uses `process.env.LANG` and `process.env.TZ` to produce this output.

I think it's moderately ok for a fallback, but an issue here is that it could lead to hydration mismatches across the server and client if the locale and time zone is different. I think we should settle for something stable.

Maybe we could reuse `dateTime` and use this:

```tsx
// Prints "1/10/2007 – 1/10/2008" in my test from above
[dateTime(start), dateTime(end)].join(' – '),
```

Another optimization on top could be to avoid computing the fallback upfront here and change this argument of `getFormattedValue` to a function to lazily execute on demand. We can also reorder the arguments of `getFormattedValue` to move the fallback function last.

Does this make sense? What do you think?

Reply:
There's been an update on the `main` branch in regard to this: `getFormattedValue` now no longer accepts a `value` but instead a `getFallback` function. You might want to rebase on the latest changes when continuing with this.